doc/developer/guide-testing.md

```diff
@@ -257,11 +259,63 @@ provide most of the coverage, but it's worth adding some (*very*) basic
 testdrive tests (e.g., `> SELECT DATE '1999-01-01'`) to ensure that our pgwire
 implementation is properly serializing dates.
 
-Like the [Unit Tests](#unitintegration-tests), many testdrive tests require
-Zookeeper, Kafka, and the Confluent Schema Registry.
+#### Usage via mzcompose/Docker
+
+The easiest way to run testdrive tests is via mzcompose.
+
+Testdrive comes with a Docker Compose environment that can set all
+of this up for you, including re-building and running the testdrive
+and materialized binaries if you have changed them locally. From the
+`test/testdrive` directory:
+
+```
+./mzcompose down -v
+BUILD_MODE=debug TD_TEST=*.td ./mzcompose run local
+```
+
+The `down` subcommand is needed to destroy any existing materialize
+containers with possibly old binaries. If this is not run, then even
+though a new materialize binary is built, it will not be used if there's
+an existing container.
+
+Supported **environment variables** (in addition to `BUILD_MODE` documented elsewhere):
+
+* `TD_TEST` (default `*.td`) is a glob of tests to run from the test/testdrive directory. The
+  default is to not run any of the "esoteric" tests.
+
+  ```
+  TD_TEST=joins.td BUILD_MODE=debug ./mzcompose run local
+  ```
+
+Supported **workflows** (target of `mzcompose run <workflow>`):
+
+* `local`: Disable AWS support
+
+  ```console
+  $ TD_TEST=*.td ./mzcompose run local
+  ```
+
+* `local-aws`: Use [LocalStack][] as an AWS stub. Use this when you want to run the esoteric tests
+  but don't have AWS credentials:
+
+  ```console
+  $ TD_TEST=esoteric/*.td ./mzcompose run local-aws
+  ```
 
-For testdrive tests that interacts with Amazon Kinesis, you will need to be
-running [LocalStack](https://github.com/localstack/localstack). To install and
+* `ci`: Expect actual AWS credentials to be available, either in the process environment or via
```

Comment:
Do we have docs anywhere about how to create AWS EC2 profiles locally? (Definitely out of scope for this PR)

Reply:
yes! updating this.